qa_eval/answer_evaluation.py

```diff
@@ -0,0 +1,70 @@
+import csv
+from pathlib import Path
+
+from openai import OpenAI
+from tqdm import tqdm
+
+
+DATA_FILE_PATH = '../data/knowledge-hub/data.tsv'
+PROMPT_FILE_PATH = 'prompts/template.md'
+OUT_FILE_PATH = 'results/knowledge-hub.tsv'
+OUT_FIELDS = ['#T', '#P', '#TP', 'Reasoning', 'Error']
+LLM_MODEL = 'gpt-4o-mini'
+TEMPERATURE = 0.0
+
+
+def call_llm(openai_client: OpenAI, prompt: str) -> str:
+    try:
+        response = openai_client.chat.completions.create(
+            model=LLM_MODEL,
+            messages=[{'role': 'user', 'content': prompt}],
+            temperature=TEMPERATURE
+        )
+        return response.choices[0].message.content.strip('\n')
+    except Exception as e:
+        return str(e).replace('\n', '    ')
+
+
+def extract_response_values(response: str) -> tuple[str, str, str, str, str]:
+    vals = response.split('\t')
+    n = len(vals)
+    if n < 4:
+        msg = f'Expected 4 tab-separated values: {response}'
+        return '', '', '', '', msg
+    vals = vals[:4]
+    try:
+        t, p, tp = map(int, vals[:3])
+    except ValueError:
+        msg = f'Non-int value: {response}'
+        return '', '', '', vals[3], msg
+    if not (1 <=t and 1 <= p and 0 <= tp <= min(t, p)):
+        msg = f'Invalid int values: {t}\t{p}\t{tp}'
+        return '', '', '', vals[3], msg
+    return vals[0], vals[1], vals[2], vals[3], ''
+
+
+def evaluate_answers():
```

Comment:
On a second look now I'm thinking the following: In order for the users of this library to use this, they have to clone this repository, which might be ok, but why don't we make this function accept arguments for DATA_FILE_PATH, PROMPT_FILE_PATH, etc., so that users can add this library as a dependency to their project(s) and use this function in their own scripts? Even if we don't do this and allow this function to be called only if the git repository is cloned, then why we don't make these DATA_FILE_PATH, PROMPT_FILE_PATH, etc. arguments to the script? Instead of saying in the instructions
```
Modify file `answer_evaluation.py`, to set variable `DATA_FILE_PATH` to the path of the input TSV file
```